src/OrleansCodeGenerator/TaskUtility.cs

```diff
@@ -0,0 +1,111 @@
+and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so,
+subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS
+OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+
+namespace OrleansCodeGenerator
+{
+    using System;
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// Extension methods for <see cref="Task"/>.
+    /// </summary>
+    public static class TaskUtility
+    {
+        /// <summary>
+        /// Returns a faulted task.
+        /// </summary>
+        /// <param name="exception">The exception which the return task faulted with.</param>
+        /// <returns>A faulted task.</returns>
+        public static Task<object> Faulted(Exception exception)
+        {
+            var completion = new TaskCompletionSource<object>();
+            completion.TrySetException(exception);
+            return completion.Task;
+        }
+
+        /// <summary>
+        /// Returns a completed task.
+        /// </summary>
+        /// <returns>A completed task.</returns>
+        public static Task<object> Completed()
+        {
+            return Task.FromResult((object)default(string));
+        }
+
+        /// <summary>
+        /// Returns a <see cref="Task{Object}"/> for the provided <paramref name="task"/>.
+        /// </summary>
+        /// <param name="task">
+        /// The task.
+        /// </param>
+        /// <returns>
+        /// The response.
+        /// </returns>
+        public static Task<object> Box(this Task task)
+        {
+            return task.ContinueWith(
+                antecedent =>
+                {
+                    if (antecedent.Exception != null)
+                    {
+                        throw new AggregateException(antecedent.Exception);
+                    }
+
+                    return (object)default(string);
+                },
+                TaskContinuationOptions.ExecuteSynchronously);
+        }
+
+        /// <summary>
+        /// Returns a <see cref="Task{Object}"/> for the provided <paramref name="task{T}"/>.
+        /// </summary>
+        /// <typeparam name="T">
+        /// The underlying type of <paramref name="task"/>.
+        /// </typeparam>
+        /// <param name="task">
+        /// The task.
+        /// </param>
+        /// <returns>
+        /// The response.
+        /// </returns>
+        public static Task<object> Box<T>(this Task<T> task)
```

Comment:
Looks similar to `OrleansTaskExtension.ConvertTaskViaTcs<T>` but I'm not 100% sure they are identical.